org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/json/adapters/MessageTypeAdapterFactory.java

```diff
@@ -88,12 +96,7 @@ public Message read(JsonReader in) throws IOException {
 					break;
 				}
 				case "params": {
-					Type parameterType = getParameterType(method);
-					if (isNullOrVoidType(parameterType)) {
-						params = new JsonParser().parse(in);
-					} else {
-						params = gson.fromJson(in, parameterType);
-					}
+					rawParams = new JsonParser().parse(in);
```

Comment:
With this change we always create a JsonElement tree, which is less efficient than creating the parameters directly from the stream. Can we still support both without too much code overhead?